eoc_journal/base_api_client.py

```diff
@@ -0,0 +1,37 @@
+"""
+A base client for API integrations.
+"""
+from abc import ABCMeta
+from django.conf import settings
+
+from .utils import build_jwt_edx_client
+
+
+# pylint: disable=R0903
+class BaseApiClient(object):
+    """
+    Base API client abstract class.
+    """
+    __metaclass__ = ABCMeta
```

Comment:
Not sure why this is needed? I think you're just using simple inheritance.

Reply:
Ah, right, I'll remove it. I was thinking about enforcing invocation of `connect()` with metaclasses, but actually I came up with more elegant solution.

Reply:
I could understand that in a library if you don't know what's going to end up subclassing it, but I think it's a bit of an overkill here! There are unlikely to be a dozen more subclasses here, so it's OK to just type `self.connect()` a few times since it keeps things easier to understand.

Reply:
By overkill you mean metaclass check or my current solution (named argument in `__init__`)?
If the first - yeah, I also though it would be an overkill, so I decided to do take a different approach :D

Reply:
The metaclass approach. This seems prefectly fine to me.